corgidrp/polflat.py

```diff
@@ -0,0 +1,377 @@
+import numpy as np
+from scipy import interpolate
+from scipy.ndimage import median_filter
+from scipy.ndimage import gaussian_filter as gauss
+from scipy import ndimage
+from scipy.signal import convolve2d
+import astropy.io.fits as fits
+from astropy.convolution import convolve_fft
+import photutils.centroids as centr
+from photutils.aperture import CircularAperture
+import corgidrp.mocks as mocks
+import corgidrp.data as data
+from astropy.stats import sigma_clipped_stats
+from photutils.detection import DAOStarFinder
+import corgidrp.flat as flat
+from skimage.transform import downscale_local_mean
+
+                                     
+def create_spatial_pol(dataset,nr=60,pfov_size=174,image_center_x=512,image_center_y=512,separation_diameter_arcsec=7.5,alignment_angle_WP1=0,alignment_angle_WP2=45,planet=None,band=None,prism='POL0'):
+    """Turns a dataset of neptune or uranus images with single planet images into the images observed through the wollaston prisms also incorporates the spatial variation of polarization on the 
+        surface of the planet
+
+    
+        Args:
+            dataset (corgidrp.data.Dataset): a dataset of image frames that are raster scanned (L2a-level)
+            nr (int): planet radius
+            pfov_size (int): size of the image created for the polarization variation, typically ~ 2 x nr
+            image_center_x (int): x coordinate of the center pixel of the final image with two orthogonal pol components
+            image_center_y (int): y coordinate of the center pixel of the final image with two orthogonal pol components
+            separation_diameter_arcsec (float): separation between the two orthogonal pol components in arcsecs
+            alignment_angle_WP1 (int): wollaston prism angle for Pol0 - 0
+            alignment_angle_WP2 (int): wollaston prism angle for Pol45- 45
+            planet (str): neptune or uranus
```

Comment:
should the code throw an error if the user does not supply neptune/uranus and the band 1/4, or should the code automatically determine neptune/uranus and band 1/4 using the header keywords in the dataset object?